boot/cmdline.go

```diff
@@ -242,3 +243,59 @@ func observeSuccessfulCommandLineCompatBoot(model *asserts.Model, m *Modeenv) (*
 	newM.CurrentKernelCommandLines = bootCommandLines{cmdlineExpected}
 	return newM, nil
 }
+
+// observeCommandLineUpdate observes a pending kernel command line change caused
+// by an update of boot config. When needed, the modeenv is updated with a
+// candidate command line and the encryption keys are resealed. This helper
+// should be called right before updating the managed boot config.
+func observeCommandLineUpdate(model *asserts.Model) error {
+	// TODO:UC20: consider updating a recovery system command line
+
+	m, err := loadModeenv()
+	if err != nil {
+		return err
+	}
+
+	if len(m.CurrentKernelCommandLines) == 0 {
+		return fmt.Errorf("internal error: current kernel command lines is unset")
+	}
+	// this is the current expected command line which was recorded by
+	// bootstate
+	cmdline := m.CurrentKernelCommandLines[0]
```

Comment:
to be clear, the reason that it is valid to always use `[0]` here is because `observeCommandLineUpdate` is only called by `UpdateManagedBootConfigs`, which is itself will only be called after or before (but never during) a kernel or gadget update, when there could be multiple kernel command lines here. As such we will never have in-progress kernel command line updates happening from other sources via bootState20CommandLine, right?

`UpdateManagedBootConfigs` is meant to only be called when the snapd snap is being updated, which will never happen during early startup of snapd before MarkBootSuccessful, is that correct?

Reply:
also relatedly, when there is a boot config update, how will we know to do a reboot? won't we have to do a reboot for boot config ? 

I'm getting a feeling that this code might need to be part of the setNext machinery in bootstate20 with the `bootState` interface ... but maybe I'm wrong and I'm missing part of the full vision here

Reply:
I think the confusion may be caused by the fact that the initial PR #8947 had more examples of use. There was an explicit `update-managed-boot-config` task that runs as part of the snapd snap refresh.

Reply:
as discussed in the SU, I had forgotten about the original implementation (even though I reviewed it!) and this sort of implementation sounds fine to me, my main concern was about making sure that this code was never run concurrently with other boot updates, etc. which in the original proposal makes sense to me, so this is all fine I think